src/arr/compiler/resolve-scope.arr

```diff
@@ -231,8 +387,10 @@ fun desugar-scope-block(stmts :: List<A.Expr>, binding-group :: BindingGroup) ->
           add-letrec-bind(binding-group, A.s-letrec-bind(
               l,
               A.s-bind(l, false, A.s-name(l, name), A.a-blank),
-              A.s-lam(l, name, params, args, ann, doc, body, _check-loc, _check, blocky)
-            ), rest-stmts)
+              # NOTE(Ben): deliberately keeping this as an s-fun;
+              # it'll get turned into an s-lam in weave-contracts
```

Comment:
Not quite; that's too harsh.  We want to be able to say
```
G :: Number
G = 12345
```
and ensure that `G` turns out ok.  But for arrow contracts, you'd wanted there to be a distinction between `lam` and `fun` in terms of how thoroughly the annotation is woven in.  Currently, we weave the annotation onto the let binding, meaning "we'll check that it's a Function, but nothing more detailed than that".  But for literal `fun` declarations, we need to preserve that information until `weave-contracts`